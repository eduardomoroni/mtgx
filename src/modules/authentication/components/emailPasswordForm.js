// @flow

import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import I18n from 'react-native-i18n'
import { View } from 'react-native'
import { Input, Button } from 'nachos-ui'
import { styles } from './styles/emailPasswordForm.style'

export class EmailPasswordForm extends Component {
  state = {
    email: '',
    password: ''
  }

  static propTypes = {
    buttonText: PropTypes.string.isRequired,
    buttonCallback: PropTypes.func.isRequired
  }

  render () {
    const { buttonCallback, buttonText } = this.props
    const { email, password } = this.state

    return (
      <View>
        <Input
          icon='ios-mail-outline'
          style={[styles.input]}
          placeholder={I18n.t('EMAIL')}
          value={this.state.email}
          onChangeText={email => this.setState({email})}
        />
        <Input
          icon='ios-lock-outline'
          style={[styles.input]}
          placeholder={I18n.t('PASSWORD')}
          value={this.state.password}
          onChangeText={password => this.setState({password})}
        />
        <Button
          kind='squared'
          type='success'
          style={[styles.button]}
          onPress={() => buttonCallback(email, password)}
        >
          {buttonText}
        </Button>
      </View>
    )
  }
}
