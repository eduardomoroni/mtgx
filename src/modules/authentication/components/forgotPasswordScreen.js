// @flow

import React, { Component } from 'react'
import I18n from 'react-native-i18n'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { H1, Button, Input } from 'nachos-ui'
import { styles } from './styles/forgotPassword.style'

export class ForgotPasswordScreen extends Component {
  state = {
    email: ''
  }

  static propTypes = {
    resetPassword: PropTypes.func.isRequired
  }

  onForgotPasswordPressed = async () => {
    this.props.resetPassword(this.state.email)
  }

  render () {
    return (
      <View style={styles.container}>
        <H1>{I18n.t('RESET_PASSWORD')}</H1>
        <Input
          icon='ios-mail-outline'
          style={[styles.input]}
          placeholder={I18n.t('EMAIL')}
          value={this.state.email}
          onChangeText={email => this.setState({email})}
        />
        <Button
          kind='squared'
          type='success'
          style={[styles.button]}
          onPress={this.onForgotPasswordPressed}
        >
          {I18n.t('RESET_PASSWORD')}
        </Button>
      </View>
    )
  }
}
