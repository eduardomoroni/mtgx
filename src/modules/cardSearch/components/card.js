// @flow

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'

import { placeholdersToSymbols } from '../../../services/realm/conversion/placeholder' // TODO: REVISIT
import { cardType, printingsType } from '../types/cardType'
import { styles } from './styles/card.styles'

// TODO: REVISIT
export const getLastPrinting = (printings: printingsType) => {
  const keys = Object.keys(printings)
  const lastPrinting = printings[keys[keys.length - 1]]
  return lastPrinting.printing
}

export class Card extends Component {
  static propTypes = {
    showCardText: PropTypes.bool,
    card: cardType
  }

  renderCardNameAndMana = () => {
    const { name, manaCost } = this.props.card

    return (
      <View style={styles.lineContainer}>
        <View style={styles.leftWord}>
          <Text key='cardName' style={styles.cardNameText}>
            {name}
          </Text>
        </View>
        <View style={styles.rightWord}>
          <Text key='cardManaCost' style={styles.mana}>
            {placeholdersToSymbols(manaCost)}
          </Text>
        </View>
      </View>
    )
  }

  renderCardTypeAndEdition = () => {
    const { type, printings } = this.props.card

    return (
      <View style={styles.lineContainer}>
        <View style={styles.leftWord}>
          <Text key='cardType' style={styles.cardTypeText}>
            {type}
          </Text>
        </View>
        <View style={styles.rightWord}>
          <Text key='cardEdition'>
            {getLastPrinting(printings)}
          </Text>
        </View>
      </View>
    )
  }

  renderTextAndPower = () => {
    const { card, showCardText } = this.props
    const { text, power, toughness } = card

    return (
      <View style={styles.cardTextContainer}>
        <View style={styles.cardText}>
          <Text key='cardText' style={styles.text}>
            {showCardText ? text : ''}
          </Text>
        </View>
        <View style={styles.cardPower}>
          <Text key='cardPowerAndToughness' style={styles.cardPowerToughness}>
            {power || toughness ? `${power}/${toughness}` : ''}
          </Text>
        </View>
      </View>
    )
  }

  render () {
    return (
      <View style={styles.container}>
        {this.renderCardNameAndMana()}
        {this.renderCardTypeAndEdition()}
        {this.renderTextAndPower()}
      </View>
    )
  }
}
