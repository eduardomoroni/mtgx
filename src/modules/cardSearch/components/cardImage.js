// @flow

import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { View, Image, ActivityIndicator } from 'react-native'
import { styles } from './styles/cardImage.styles'

export class CardImage extends PureComponent {
  static propTypes = {
    multiverseid: PropTypes.number
  }

  state = { loading: true }

  renderSpinner = () => {
    return this.state.loading ? <ActivityIndicator style={styles.spinner} size='large' /> : <View />
  }

  render () {
    const { multiverseId, style } = this.props
    const IMG_URL = `http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=${multiverseId}&type=card`

    return (
      <View style={style || styles.container}>
        <Image
          resizeMode='contain'
          resizeMethod='scale'
          style={styles.image}
          source={{uri: IMG_URL}}
          onLoad={() => this.setState({loading: false})}
          onError={() => console.error('Error while loading card image')} />
        {this.renderSpinner()}
      </View>
    )
  }
}
