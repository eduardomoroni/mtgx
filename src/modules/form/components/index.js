// @flow

export * from './inputLabel'
export * from './inputPicker'
export * from './textInputForm'
export * from './switchInputForm'
export * from './modalToggle'
export * from './multiSelect'
// $FlowFixMe
export * from './numericInputForm'
// $FlowFixMe
export * from './dropdownInputForm'
