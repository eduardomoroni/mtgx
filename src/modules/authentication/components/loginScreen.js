// @flow

import React, { Component } from 'react'
import I18n from 'react-native-i18n'
import PropTypes from 'prop-types'
import { View, TouchableOpacity } from 'react-native'
import { H1, Strong } from 'nachos-ui'
import { styles } from './styles/loginScreen.style'
import { EmailPasswordForm } from './emailPasswordForm'

export class LoginScreen extends Component {
  static propTypes = {
    logUserIn: PropTypes.func.isRequired
  }

  onSignInPressed = async (email: string, password: string) => {
    this.props.logUserIn(email, password)
  }

  onSignUpPressed = () => {
    this.props.navigator.push({screen: 'authentication.signup'})
  }

  onForgotPasswordPressed = () => {
    this.props.navigator.push({screen: 'authentication.forgotpassword'})
  }

  render () {
    return (
      <View style={styles.container}>
        <H1>{I18n.t('LOGIN')}</H1>
        <EmailPasswordForm
          buttonText={I18n.t('LOGIN')}
          buttonCallback={this.onSignInPressed}
        />
        <View style={styles.footer}>
          <TouchableOpacity onPress={this.onForgotPasswordPressed}>
            <Strong style={[styles.strong]}>
              {I18n.t('FORGOT_YOUR_PASSWORD?')}
            </Strong>
          </TouchableOpacity>
          <TouchableOpacity onPress={this.onSignUpPressed}>
            <Strong style={[styles.strong]}>
              {I18n.t('DONT_HAVE_ACCOUNT?')}
            </Strong>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}
