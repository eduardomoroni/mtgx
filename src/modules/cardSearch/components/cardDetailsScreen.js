// @flow

import I18n from 'react-native-i18n'
import React, { Component } from 'react'
import { Text, View, ScrollView } from 'react-native'

import { styles } from './styles/cardDetailsScreen.styles'
import { CardImage } from './cardImage'
import { cardType } from '../types/cardType'

export class CardDetailsScreen extends Component {
  static propTypes = {
    card: cardType
  }

  renderRow = (rowName: string, rowText: string) => {
    if (rowText === null) {
      return null
    }

    return (
      <View style={styles.rowContainer}>
        <Text style={styles.rowText}>
          <Text style={styles.rowName}>{`${I18n.t(rowName)}: `}</Text>
          {rowText}
        </Text>
      </View>
    )
  }

  render () {
    const { card } = this.props

    return (
      <ScrollView style={styles.container}>
        <CardImage style={styles.card} multiverseId={card.multiverseid} />
        {this.renderRow('CARD_NAME', card.name)}
        {this.renderRow('TYPE', card.type)}
        {this.renderRow('POWER_TOUGHNESS', `${card.power}/${card.toughness}`)}
        {this.renderRow('MANA_COST', `${card.manaCost} (${card.cmc})`)}
        {this.renderRow('RARITY', card.rarity)}
        {this.renderRow('CARD_TEXT', card.text)}
        {this.renderRow('FLAVOR_TEXT', card.flavor)}
      </ScrollView>
    )
  }
}
