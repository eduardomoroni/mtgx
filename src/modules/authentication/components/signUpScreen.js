// @flow

import React, { Component } from 'react'
import I18n from 'react-native-i18n'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { H1 } from 'nachos-ui'
import { styles } from './styles/loginScreen.style'
import { EmailPasswordForm } from './emailPasswordForm'

export class SignUpScreen extends Component {
  static propTypes = {
    signUpUser: PropTypes.func.isRequired
  }

  onSignUpPressed = async (email: string, password: string) => {
    this.props.signUpUser(email, password)
  }

  render () {
    return (
      <View style={styles.container}>
        <H1>{I18n.t('SIGNUP')}</H1>
        <EmailPasswordForm
          buttonText={I18n.t('SIGNUP')}
          buttonCallback={this.onSignUpPressed}
        />
      </View>
    )
  }
}
