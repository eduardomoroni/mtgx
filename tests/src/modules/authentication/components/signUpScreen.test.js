// @flow

import React from 'react'
import { shallow } from 'enzyme'
import { SignUpScreen } from '../../../../../src/modules/authentication/components/signUpScreen'
import { EmailPasswordForm } from '../../../../../src/modules/authentication/components/emailPasswordForm'

const props = {
  signUpUser: jest.fn()
}

describe('<SignUpScreen />', () => {
  const wrapper = shallow(<SignUpScreen {...props} />)

  it('should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should pass signUp function as callback to EmailPasswordForm', () => {
    let email = 'email'
    let password = 'password'
    const form = wrapper.find(EmailPasswordForm)
    const onSignUpPressed = wrapper.instance().onSignUpPressed

    onSignUpPressed(email, password)
    expect(props.signUpUser).toHaveBeenCalledWith(email, password)
    expect(onSignUpPressed).toEqual(form.prop('buttonCallback'))
  })
})
