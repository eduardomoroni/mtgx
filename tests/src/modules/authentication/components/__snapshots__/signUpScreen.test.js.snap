// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpScreen /> should have a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <H1>
    [missing "en.SIGNUP" translation]
  </H1>
  <EmailPasswordForm
    buttonCallback={[Function]}
    buttonText="[missing \\"en.SIGNUP\\" translation]"
  />
</View>
`;
